scaleway/resource_instance_ip_test.go

```diff
@@ -125,6 +126,182 @@ func TestAccScalewayInstanceIP_Tags(t *testing.T) {
 	})
 }
 
+func TestAccScalewayInstanceIP_RoutedMigrate(t *testing.T) {
+	tt := NewTestTools(t)
+	defer tt.Cleanup()
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:      testAccCheckScalewayInstanceIPDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: `
+						resource "scaleway_instance_ip" "main" {
+						}
+					`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayInstanceIPExists(tt, "scaleway_instance_ip.main"),
+					resource.TestCheckResourceAttr("scaleway_instance_ip.main", "type", "nat"),
+				),
+			},
+			{
+				Config: `
+						resource "scaleway_instance_ip" "main" {
+						}
+						resource "scaleway_instance_ip" "copy" {
+						}
+					`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayInstanceIPExists(tt, "scaleway_instance_ip.main"),
+					testAccCheckScalewayInstanceIPExists(tt, "scaleway_instance_ip.copy"),
+					resource.TestCheckResourceAttr("scaleway_instance_ip.main", "type", "nat"),
+					resource.TestCheckResourceAttr("scaleway_instance_ip.copy", "type", "nat"),
+					resource.TestCheckResourceAttrPair("scaleway_instance_ip.main", "id", "scaleway_instance_ip.copy", "id"),
+				),
+				ResourceName: "scaleway_instance_ip.copy",
+				ImportState:  true,
+				ImportStateIdFunc: func(state *terraform.State) (string, error) {
+					return state.RootModule().Resources["scaleway_instance_ip.main"].Primary.ID, nil
+				},
+				ImportStatePersist: true,
+			},
+			{
+				Config: `
+						resource "scaleway_instance_ip" "main" {
+							type = "routed_ipv4"
+						}
+						resource "scaleway_instance_ip" "copy" {
+						}
+					`,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayInstanceIPExists(tt, "scaleway_instance_ip.main"),
+					testAccCheckScalewayInstanceIPExists(tt, "scaleway_instance_ip.copy"),
+					resource.TestCheckResourceAttr("scaleway_instance_ip.main", "type", "routed_ipv4"),
+					resource.TestCheckResourceAttrPair("scaleway_instance_ip.main", "id", "scaleway_instance_ip.copy", "id"),
+				),
+			},
+			{
+				// After the main IP migrated, we check that there is no ForceNew on the copy
```

Comment:
Is the conversion instantaneous? Do we have a waiting mechanism here in case it is not?

Reply:
Conversion is instantanous